### PR TITLE
Testbench feature merge

### PR DESCRIPTION
**Pull request by gr455**
Merge request for testbench feature

### CODE REVIEW

**Comment by gr455** [comment]
> Code Climate has analyzed commit [c3954b9](https://github.com/CircuitVerse/CircuitVerse/commit/c3954b9ab8e61799958fee9a80f79c9eb94e3702) and detected **691 issues** on this pull request.
> 
> Here's the issue category breakdown:
> 
> Category	Count
> Complexity	12
> Duplication	30
> Style	2
> Clarity	450
> Bug Risk	114
> Compatibility	83
> View more on [Code Climate](https://codeclimate.com/github/CircuitVerse/CircuitVerse/pull/2742).

Merry Christmas

File: app/views/simulator/edit.html.erb [hunk 1]
```diff
@@ -222,6 +222,69 @@
...
+        </div>
+        <span class="testbench-runall-label">
+          <span id="runall-summary">placeholder</span> Tests Passed <span id="runall-detailed-link" style="color: #18A2CD;">View Detailed</span>
+        </span>
+      </div>
+  </div>
+  <div class="panel-body tb-test-null">
+    <div class="tb-manual-test-data">
+        <div style="margin-bottom: 10px; overflow: auto;">
+          <p><i>No Test is attached to the current circuit</i></p>
```
Comment (tachyons) [inline app/views/simulator/edit.html.erb:282]: Can we make these text i18n compatiable ?

File: app/views/testbench/creator.html.erb [hunk 2]
```diff
@@ -0,0 +1,50 @@
+<!-- <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.0.1/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-+0n0xVW2eSR5OomGNYDnhzAbDsOXxcvSN1TPprVMTNDbiYZCxYbOOl7+AMvyTG2x" crossorigin="anonymous">
+<script src="https://cdn.jsdelivr.net/npm/bootstrap@5.0.1/dist/js/bootstrap.bundle.min.js" integrity="sha384-gtEjrD/SeCtmISkJkNUaaKMoLD0//ElJ19smozuHV6z3Iehds+3Ulb9Bn9Plx0x4" crossorigin="anonymous"></script> -->
+<!-- <script src="https://code.jquery.com/jquery-3.6.0.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script> -->
+<script src="https://code.jquery.com/ui/1.12.1/jquery-ui.js"></script>
```
Comment (tachyons) [inline app/views/testbench/creator.html.erb:4]: Why custom jquery ui here?

File: simulator/src/ux.js [hunk 3]
```diff
@@ -556,10 +557,21 @@ export function setupPanels() {
...
     $('.timing-diagram-panel .minimize').trigger('click');
 
+    // Update the Testbench Panel UI
+    updateTestbenchUI();
+    // Minimize Testbench UI
+    $('.testbench-manual-panel .minimize').trigger('click');
+
+    // Hack because minimizing panel then maximizing sets visibility recursively
+    // updateTestbenchUI calls some hide()s which are undone by maximization
+    // TODO: Remove hack
```
Comment (tachyons) [inline simulator/src/ux.js:572]: Create a github issue track this hack

File: app/views/testbench/creator.html.erb (hunk 2 above)
Comment (gr455) [reply app/views/testbench/creator.html.erb:4]: Yeah I will have to remove it, it was from back when the view was served statically

File: app/views/layouts/testbench.html.erb [hunk 4]
```diff
@@ -0,0 +1,20 @@
+<html>
+  <head>
+    <title>CircuitVerse - Digital Circuit Simulator online</title>
+    <%= csrf_meta_tags %>
+    <script>window.locale = "<%= I18n.locale %>"</script>
+    <script src="/js/pace.js"></script>
+    <script src="https://code.jquery.com/jquery-3.6.0.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script>
+    <script src="https://code.jquery.com/ui/1.12.1/jquery-ui.js"></script>
```
Comment (tachyons) [inline app/views/layouts/testbench.html.erb:8]: Can we use the jquery from npm?

**Review by tachyons — APPROVED** [review]

**Comment by gr455** [comment]
Rest of the linting issues are mostly use-after-defines and camel case on loop variables which I feel, is better done the way I have done.

**Comment by gr455** [comment]
@tachyons This can be merged now if all looks good

**Comment by tachyons** [comment]
@gr455 I pushed a commit to update configuration. Can you please check if app works as expected now ?

**Comment by gr455** [comment]
> @gr455 I pushed a commit to update configuration. Can you please check if app works as expected now ?

Yes I will check :+1:

**Comment by gr455** [comment]
It works properly now, we can finally merge this @tachyons

File: app/views/testbench/creator.html.erb (hunk 2 above)
Comment (tachyons) [reply app/views/testbench/creator.html.erb:4]: @gr455 Any update on this ?

File: simulator/src/css/main.stylesheet.css [hunk 5]
```diff
@@ -1247,6 +1456,23 @@ font: inherit;
 
 /*! combinationalAnalysis dialog styling end here */
 
+#setTestbenchData {
+  
+}
```
Comment (tachyons) [inline simulator/src/css/main.stylesheet.css:1461]: Empty rule ?

File: app/views/testbench/creator.html.erb (hunk 2 above)
Comment (gr455) [reply app/views/testbench/creator.html.erb:4]: Whoops completely forgot about this. Will add it to asset pipeline

**Pull request merged**